### ISSUE TITLE
cblas_dgemm returns incorrect results when the static feature of openblas-src is enabled on macOS

### ISSUE THREAD

**Issue by maoe** [blas-lapack-rs/openblas-src#64]
https://github.com/maoe/ndarray-dot-repro is a reproducer. It was minimized from code which used argmin + ndarrayl feature.

This reproducer uses the patched version of `openblas-src` from https://github.com/blas-lapack-rs/openblas-src/pull/65 so that static build works both on Linux and macOS Big Sur.

The expected output of the main is `[1.0, 1.0, 1.0, 1.0]` but it returns the zero matrix on macOS when the `static` feature of `openblas-src` is enabled.

```
% cargo run
   Compiling ndarray-dot-repro v0.1.0 (/Users/maoe/src/github.com/maoe/ndarray-dot-repro)
    Finished dev [unoptimized + debuginfo] target(s) in 0.49s
     Running `target/debug/ndarray-dot-repro`
[
    1.0,
    1.0,
    1.0,
    1.0,
]
% cargo run --features openblas-src/static
   Compiling ndarray-dot-repro v0.1.0 (/Users/maoe/src/github.com/maoe/ndarray-dot-repro)
    Finished dev [unoptimized + debuginfo] target(s) in 0.52s
     Running `target/debug/ndarray-dot-repro`
[
    0.0,
    0.0,
    0.0,
    0.0,
]
```

FWIW, I haven't been able to reproduce the issue in C on macOS:

```console
% git clone https://github.com/xianyi/OpenBLAS.git
% cd OpenBLAS
% git checkout v0.3.14 # This is the commit #65 uses
% make
% wget -q https://gist.githubusercontent.com/maoe/953c4e863703be915928b0f8997fcc46/raw/e827d61ae2944a72a37d0784c0aab0026e1b6cdb/repro.c # See https://gist.github.com/maoe/953c4e863703be915928b0f8997fcc46
% cc -o repro repro.c libopenblas.a
% ./repro
1.000000
1.000000
1.000000
1.000000
```

**Comment by maoe** [comment]
https://github.com/xianyi/OpenBLAS/issues/1088 looks similar but I'm not sure if it's the same issue as I haven't reproduced it in C yet.

**Comment by maoe** [comment]
Removed the dependency on `blas-src` on current master of the reproducer.

**Comment by maoe** [comment]
FWIW this issue remains even with #65.

**Comment by maoe** [comment]
Updated the reproducer with #65 and a simpler matrix input and updated the description accordingly.

**Comment by maoe** [comment]
I managed to fix the issue with https://github.com/maoe/ndarray-dot-repro/commit/4d94c7357183d11a826d3ddf8886f9d6cb256fc0.

Apparently, at least on macOS, the static feature needs an empty extern block with `#[link(name = "openblas")]` attribute to work. I also tried putting the block in `openblas-src`'s `lib.rs` but it didn't work. It seems that the block needs to be at the use site.

I guess [the instructions on the wiki](https://github.com/blas-lapack-rs/blas-lapack-rs.github.io/wiki) need updating.
